base/file.jl

```diff
@@ -273,30 +277,37 @@ Stacktrace:
 function rm(path::AbstractString; force::Bool=false, recursive::Bool=false)
     if islink(path) || !isdir(path)
         try
-            @static if Sys.iswindows()
-                # is writable on windows actually means "is deletable"
-                st = lstat(path)
-                if ispath(st) && (filemode(st) & 0o222) == 0
-                    chmod(path, 0o777)
-                end
-            end
             unlink(path)
         catch err
-            if force && isa(err, IOError) && err.code==Base.UV_ENOENT
-                return
+            if isa(err, IOError)
+                force && err.code==Base.UV_ENOENT && return
+                @static if Sys.iswindows()
+                    if err.code==Base.UV_EACCES && endswith(path, ".dll")
+                        # Loaded DLLs cannot be deleted on Windows, even with posix delete mode
+                        # but they can be moved. So move out to allow the dir to be deleted
+                        # TODO: Add a mechanism to delete these moved files after dlclose or process exit
```

Comment:
Should this TODO instead reference Pkg.gc()?

Reply:
I'm going to fix it in the pkg bump PR

Reply:
What happens if a file exists already in the directory with the same name?

Reply:
I assumed using tempname makes the chances of that low enough to not be an issue? I can add a loop